src/src/components/TodoItem.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from '../context/TodosContext';
+
+type Props = {
+  todo: Todo
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+}) => {
+  const {
+    deleteTodo,
+    upgradeTodo,
+    chosenTodoId,
+  } = useContext(TodosContext);
+
+  const [isEdited, setIsEdited] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const focusTodo = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    focusTodo.current?.focus();
+  }, [isEdited]);
+
+  const handleComplete = () => {
+    upgradeTodo({
+      ...todo,
+      completed: !todo.completed,
+    });
+  };
+
+  const hendleDoubleClick = () => {
+    setIsEdited(true);
+  };
+
+  const handleUpdateTodo = () => {
+    if (!newTitle.trim()) {
+      deleteTodo(todo.id);
+
+      return;
+    }
+
+    setIsEdited(false);
+
+    upgradeTodo({
+      ...todo,
+      title: newTitle.trim(),
+    });
+  };
+
+  const handleKey = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Escape') {
+      setNewTitle(todo.title);
+      setIsEdited(false);
+    }
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={classNames(
+        'todo', {
+          completed: todo.completed,
+        },
+      )}
+      onDoubleClick={hendleDoubleClick}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked
+          onChange={() => handleComplete()}
+        />
+      </label>
+
+      {isEdited ? (
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleUpdateTodo();
+          }}
+        >
+          <input
+            data-cy="TodoTitleField"
+            type="text"
+            className="todo__title-field"
+            placeholder="Empty todo will be deleted"
+            ref={focusTodo}
+            value={newTitle}
+            onChange={event => setNewTitle(event.target.value)}
+            onBlur={() => handleUpdateTodo()}
```

Comment:
```suggestion
            onBlur={handleUpdateTodo}
```
redundant function wrapper